app/src/main/java/com/example/javaopencommerce/order/OrderIntegrityValidator.java

```diff
@@ -0,0 +1,135 @@
+package com.example.javaopencommerce.order;
+
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toUnmodifiableList;
+
+import com.example.javaopencommerce.item.ItemQueryRepository;
+import com.example.javaopencommerce.item.dtos.CardDto;
+import com.example.javaopencommerce.item.dtos.ItemDto;
+import com.example.javaopencommerce.item.dtos.ProductDto;
+import com.example.javaopencommerce.order.dtos.OrderDetailsDto;
+import com.example.javaopencommerce.order.exceptions.AddressNotProvidedValidationException;
+import com.example.javaopencommerce.order.exceptions.OrderStateValidationException;
+import com.example.javaopencommerce.order.exceptions.OrderValidationException;
+import com.example.javaopencommerce.order.exceptions.OrderValueNotMatchingValidationException;
+import com.example.javaopencommerce.order.exceptions.OutOfStockItemsValidationException;
+import io.smallrye.mutiny.Uni;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+
+class OrderIntegrityValidator {
+
+  private final ItemQueryRepository itemQueryRepository;
+  private List<OrderValidationException> orderInaccuracies;
+
+  OrderIntegrityValidator(ItemQueryRepository itemQueryRepository) {
+    this.itemQueryRepository = itemQueryRepository;
+  }
+
+  public Uni<Void> validateOrder(OrderDetailsDto order) {
+    orderInaccuracies = new ArrayList<>();
+    validateOrderBasics(order);
```

Comment:
It's risky. 
Meaning the class could easily be used by inexperienced developer in wrong way. 
Having orderInaccuracies (or errorList) as a method parameter, you could pass it to `validateOrderBasics` and `validateOrderState` to fill it. This would cause side effects though. Maybe better idea is for those methods to return error list despite `orderInaccuracies`:

```suggestion
    var orderInaccuracies = new ArrayList<OrderValidationException>();
    orderInaccuracies.addAll(validateOrderBasics(order));
```